src/components/Answer/index.tsx

```diff
@@ -7,30 +9,71 @@ import MultiInputs from 'components/MultiInputs';
 import Nps from 'components/Nps';
 import Rating from 'components/Rating';
 import TextArea from 'components/TextArea';
-import { Answer as AnswerType } from 'types/answer';
+import { Answer as AnswerType, instanceOfAnswer } from 'types/answer';
 import { DisplayType, Question, getDisplayTypeEnum } from 'types/question';
-import { AnswerRequest } from 'types/request/surveySubmitRequest';
+import { AnswerRequest, instanceOfAnswerRequest } from 'types/request/surveySubmitRequest';
 
 export const answerDataTestIds = {
   base: 'answer__base',
 };
 
 interface AnswerProps {
   question: Question;
+  onAnswerChanged: (answers: AnswerRequest[]) => void;
 }
-const Answer = ({ question }: AnswerProps): JSX.Element => {
+
+const Answer = ({ question, onAnswerChanged }: AnswerProps): JSX.Element => {
   const displayTypeEnum = getDisplayTypeEnum(question);
 
   const onValueChanged = (answer: number | AnswerType | AnswerRequest) => {
```

Comment:
There is an attribute in the answer object `input_mask` that can let you know the format or the field. Should we use it to validate the value in those fields as well?
![image](https://github.com/manh-t/react-survey/assets/7344405/a2895971-3c63-48df-af50-42f051032f04)

Reply:
@hoangmirs yeah, it's nice to have this feature. But please allow me to improve it later in [this task](https://github.com/manh-t/react-survey/issues/93) 🙏